crates/matrix-sdk/src/event_cache/linked_chunk.rs

```diff
@@ -1108,6 +1048,76 @@ where
     }
 }
 
+/// A type to listen to what's happening inside a [`LinkedChunk`].
+pub trait LinkedChunkListener<Item, Gap>: Send + Sync {
+    /// The error.
+    type Error: Error + fmt::Debug + 'static;
+
+    /// A new chunk of type `ChunkContent::Items` has been inserted.
+    fn new_chunk_items(
```

Comment:
(Maybe fold this commit with the previous one, since it moves code inserted by the previous commit?)

Also, my naming comment still holds :) (re: putting "chunk" at the end)